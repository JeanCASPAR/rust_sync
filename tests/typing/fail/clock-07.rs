use rust_sync::sync;

sync! {
    #![pass(1)]

    node oui(c : bool) = ()
    where
        a : float = merge c {
            true => (3.0 when c) when c,
            false => (2.0 when c) whennot c,
        };
}

fn main() {}
