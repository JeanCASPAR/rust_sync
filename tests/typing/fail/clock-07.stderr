error: clock error: the expression of the true branch has an incompatible clock type
 --> tests/typing/fail/clock-07.rs:7:21
  |
7 |             true => (3.0 when c) when c,
  |                     ^^^^^^^^^^^^^^^^^^^
  |
  = note: its clock type starts with `base on c`
note: the clock type of c is `base`
 --> tests/typing/fail/clock-07.rs:6:27
  |
6 |         a : float = merge c {
  |                           ^
